docs(configuration): fix tip for mode option (#4605)
webpack/webpack.js.org: src/content/configuration/mode.md

If not set, webpack sets `production` as the default value for `mode`.

> T> If `mode` is not provided via configuration or CLI, CLI will check if `NODE_ENV` is set to any of the accepted values for `mode` and will automatically use it to set `mode`.

Author: Maybe this sounds better?

```suggestion
T> If `mode` is not provided via configuration or CLI, CLI will automatically pick `mode` from `NODE_ENV` if it is set to any of the accepted values.
```

> T> If `mode` is not provided via configuration or CLI, CLI will check if `NODE_ENV` is set to any of the accepted values for `mode` and will automatically use it to set `mode`.

Reviewer: ```suggestion
T> If `mode` is not provided via configuration or CLI, CLI will use any valid `NODE_ENV` value for `mode`.
```
